### ISSUE TITLE
Allow plugging into stream feeder mechanism

### ISSUE THREAD

**Issue by fatcerberus** [liballeg/allegro5#803]
`al_register_audio_stream_loader()` currently requires you to provide it with a function that returns an `ALLEGRO_AUDIO_STREAM`, which you then have to feed manually.  This is not much more useful than simply creating the stream yourself with `al_create_audio_stream()`.  Best I can tell, you can't use the seeking functions, etc.

A better API would be for Allegro to manage and feed the stream itself and your game simply provides it with raw sample data.

**Comment by SiegeLord** [comment]
Yeah! We had an idea just like this, but it never got implemented. It'd be a different kind of audio stream, which you could read audio data from and it would plug into the current audio stream as you suggest.
